lib/logos.js

```diff
@@ -54,28 +54,56 @@ function getShieldsIcon({ name, color }) {
   }
 }
 
-function getSimpleIcon({ name, color }) {
+function brightness({ r, g, b }) {
+  return +((r * 299 + g * 587 + b * 114) / 255000).toFixed(2)
+}
+
+function hexToRgb(hex) {
+  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex)
+  return {
+    r: parseInt(result[1], 16),
+    g: parseInt(result[2], 16),
+    b: parseInt(result[3], 16),
+  }
+}
+
+function getSimpleIconColor({ icon, style }) {
+  const { hex } = icon
+  if (style !== 'social' && brightness(hexToRgb(hex)) <= 0.4) {
+    return 'whitesmoke'
+  }
+  if (style === 'social' && brightness(hexToRgb(hex)) >= 0.6) {
+    return '#333'
+  }
+  return hex
+}
+
+function getSimpleIcon({ name, color, style }) {
   const key = name.toLowerCase().replace(/ /g, '-')
 
   if (!(key in simpleIcons)) {
     return undefined
   }
 
   const { svg, base64 } = simpleIcons[key]
-  const svgColor = toSvgColor(color)
+  const svgColor =
+    toSvgColor(color) ||
+    toSvgColor(getSimpleIconColor({ icon: simpleIcons[key], style }))
   if (svgColor) {
     return svg2base64(svg.replace('<svg', `<svg fill="${svgColor}"`))
   } else {
```

Comment:
>The way this is written it looks like we will never end up here. i.e. we will always recompute the svg + the base64 transform.

Yeah, think this may be the case.

>Should we precompute the `whitesmoke` and `#333` versions when the logos are loaded, instead?

Good idea, it should save a quite a bit of resources in the long run.

and then we would just need to update the replace to something like:
```js
svg.replace(/fill=".+?"/, `fill="${svgColor}"`);
```

Although it could probably be done in a separate PR later on if its too much hassle.